toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -0,0 +1,199 @@
+package imagecustomizerlib
+
+import (
+	"archive/tar"
+	"crypto/sha512"
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImageBuildData struct {
+	Source    string
+	KnownInfo ExpectedImage
+	Metadata  *Image
+}
+
+type ExpectedImage struct {
+	Name                string
+	PartType            PartitionType
+	MountPoint          string
+	FsType              string
+	FsUuid              string
+	OsReleasePath       *string
+	GrubCfgPath         *string
+	ContainsRpmDatabase bool
+}
+
+func (ex ExpectedImage) ShouldMount() bool {
+	return ex.OsReleasePath != nil || ex.GrubCfgPath != nil || ex.ContainsRpmDatabase
```

Comment:
trident has [code](https://dev.azure.com/mariner-org/ECF/_git/trident?path=/tools/cmd/mkcosi/variants/common.go&version=GBmain&line=361&lineEnd=379&lineStartColumn=1&lineEndColumn=3&lineStyle=plain&_a=contents ) to extract this info. Do I need to add logic to mount the partition and search for os release file path or grub config path?

Reply:
That code is a temporary COSI generator (mkcosi), it's not part of trident itself. I'm not an expert on MIC internals, but from the outside it would seem easier to extract this info when MIC has the full image mounted instead of doing it after the fact, like mkcosi does. 
The mkcosi tool is just doing the best it can given the circumstances, so I don't recommend copying too much from it. :) <sup>(I wrote it with the full intention of throwing it away, so it is by no means good or well-designed code.)</sup>

Reply:
right, I excluded trident's mount logic. I should remove this function but focus on extracting the values. what is ContainsRpmDatabase and do you think it is something we can obtain?

Reply:
In mkcosi it signals that the partition contains `/var/lib/rpm` (where the RPM database is located), this should not be needed in MIC since the full mount tree is actually constructed for the chroot, so you don't really need to track which partition has what.
mkcosi's inputs are the partition images produced by MIC, I chose to be lazy and pre-define things such as `ContainsRpmDatabase` instead of doing the work of finding it because mkcosi only supports two image variants.

Reply:
In general, the whole concept of "expected image" from mkcosi may be superfluous in MIC.

Reply:
removed. Thanks!